Add %site_title% and %site_description% macros

arnaudbroes opened the pull request:
#1271.

michaeltorbert commented:
This should have unit testing for the old macros as well as the new in order to verify the behavior.

arnaudbroes commented:
We should have for each and another to compare them and see if the result is the same.

wpsmort commented:
This is tested and is good.

arnaudbroes commented:
@EkoJR reach out to me first before looking at this since I'll need to add a unit test.

EkoJR commented on aioseop_class.php:
```diff
@@ -919,6 +919,9 @@ public function get_title_format( $args ) {
 		$w            = $info['w'];
 		$p            = $info['p'];
 
+		if ( strpos( $title_format, '%site_title%' ) !== false ) {
```
The parent class method for strpos should be used instead. (multiple occurances)

EkoJR replied:
https://github.com/semperfiwebdesign/all-in-one-seo-pack/blob/53b0a5f9c74941979d645caa4866145287f7e944/admin/aioseop_module_class.php#L302

arnaudbroes replied:
Done.

EkoJR commented on aioseop_class.php:
```diff
@@ -919,6 +919,9 @@ public function get_title_format( $args ) {
+		if ( strpos( $title_format, '%site_title%' ) !== false ) {
+			$title_format = str_replace( '%site_title%', get_bloginfo( 'name' ), $title_format );
+		}
 		if ( strpos( $title_format, '%blog_title%' ) !== false ) {
```
It might help to add a dev note here that %blog_title% is deprecated.

arnaudbroes replied:
Done.

EkoJR commented on aioseop_class.php:
```diff
@@ -1462,7 +1465,13 @@ function get_aioseop_title( $post, $use_original_title_format = true ) {
 			$title_format = $aioseop_options['aiosp_search_title_format'];
-			$title        = str_replace( '%blog_title%', $this->internationalize( get_bloginfo( 'name' ) ), $title_format );
+			$title        = str_replace( '%site_title%', $this->internationalize( get_bloginfo( 'name' ) ), $title_format );
+			if ( strpos( $title, '%blog_title%' ) !== false ) {
```
The parent class method for strpos should be used instead. (multiple occurances)

Using Yoda operations is also a good habit to get into. `if ( 'value === $variable ) {`. It's a WP Standard and values are generally shorter than variable & function names. (multiple occurances)

arnaudbroes replied:
Done.

arnaudbroes commented:
@EkoJR I've addressed all comments. Now I am waiting on either you or @contactashish13 to help me finalize my PHPUnit setup so I can add the unit tests that were requested.

arnaudbroes commented:
@EkoJR this can be code-reviewed again. I implemented all the comments and added the required unit tests. I'm going to expand those unit tests to include all pages (homepage, search page, 404 page, ...) and also test all scenario's (homepage title, aioseop title, homepage as static page, etc ..)
